### PR TITLE
Dspace-OpenAIRE-Funders

### LINKED ISSUE #2816: Search/Import Funded Projects from OpenAIRE API to create new Entities
From release plan; assigned to Dimitris at OpenAIRE. 
Estimate: 60 hours
DEPENDS ON "Convert External Source to Entity" FEATURE.  Would require a new external source to support searching the OpenAIRE API

### PR DESCRIPTION
**Pull request by dimitrispie**
Fixes #2816

### CODE REVIEW

File: dspace-api/pom.xml [hunk 1]
```diff
@@ -673,7 +673,12 @@
             <artifactId>hibernate-jpa-2.1-api</artifactId>
             <version>1.0.0.Final</version>
         </dependency>
-
+        <dependency>
```
Comment (paulo-graca) [inline dspace-api/pom.xml]: This PR is far behind, currently DSpace 7 requires at least Java 11 and it's not backward compatible. This PR should be upgraded.
I'm getting this compiling error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project dspace-services: Compilation failure
[ERROR] DSpace2655_git/dspace-services/src/main/java/org/dspace/servicemanager/config/DSpaceConfigurationPlaceholderConfigurer.java:[27,8] cannot access java.lang.Object
[ERROR] bad class file: /modules/java.base/java/lang/Object.class
```

File: dspace-api/src/main/java/org/dspace/external/provider/impl/OpenAIREFunderDataProvider.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        ExternalDataObject externalDataObject = convertToExternalDataObject(response);
+        return Optional.of(externalDataObject);
+    }
+
+    protected ExternalDataObject convertToExternalDataObject(Response response) {
+        ExternalDataObject externalDataObject = new ExternalDataObject(sourceIdentifier);
+        Results results = response.getResults();
+
+        Result result = results.getResult();
+        Project project = response.getResults().getResult().getMetadata().getEntity().getProject();
```
Comment (paulo-graca) [inline dspace-api/src/main/java/org/dspace/external/provider/impl/OpenAIREFunderDataProvider.java]: With this kind of chained gets, we might requesting data on NULL objects. I think the case of not having data, or parts of it, should be addressed. Perhaps throwing exceptions to deal with this later.

File: dspace-api/src/main/java/org/dspace/external/provider/impl/OpenAIREFunderDataProvider.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+/**
+ * This class is the implementation of the ExternalDataProvider interface that
+ * will deal with the OpenAIRE API Funders Data lookup
+ */
+public class OpenAIREFunderDataProvider implements ExternalDataProvider {
+
+    private static Logger log = LogManager.getLogger(OpenAIREFunderDataProvider.class);
+
+    private String sourceIdentifier;
```
Comment (paulo-graca) [inline dspace-api/src/main/java/org/dspace/external/provider/impl/OpenAIREFunderDataProvider.java]: can you add java docs to describe this properties?

File: dspace-api/src/main/java/org/dspace/external/provider/impl/OpenAIREFunderDataProvider.java [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    }
+
+    @Override
+    public boolean supports(String source) {
+        return StringUtils.equalsIgnoreCase(sourceIdentifier, source);
+    }
+
+    @Override
+    public int getNumberOfResults(String query) {
+        return 1;
```
Comment (paulo-graca) [inline dspace-api/src/main/java/org/dspace/external/provider/impl/OpenAIREFunderDataProvider.java]: 1 ???

**Review by paulo-graca — CHANGES_REQUESTED** [review]
Thank you @dimitrispie for this contribution. Looking at the code, it's very simple to review. I couldn't test it perhaps because of my java 11 ready system. My comment and major concern is related with the need to deal with lack or missing data which are missing in this PR.

**Comment by tdonohue** [comment]
This work/PR appears to be outdated / possibly "stalled".  I just wanted to note that #2816 (the ticket associated with this PR) was/is scheduled for 7.0 beta 5 (which is starting development now).  Therefore, @dimitrispie if this PR is able to be updated/cleaned up, I can work on getting additional reviewers.  If not, I may need to reschedule this work for 7.1 (as beta 5 will be the final beta prior to 7.0).

**Comment by paulo-graca** [comment]
@tdonohue, @dimitrispie created this other https://github.com/DSpace/DSpace/pull/2987 Pull Request to address the changes. 
I've already ask him to to a recent pull since the code seams outdated in that PR.

**Comment by paulo-graca** [comment]
Hi @dimitrispie , the source repository for this PR isn't available. I think it will be no longer possible to merge the code. Does it makes sense to close this PR?

**Review by paulo-graca — CHANGES_REQUESTED** [review]
The source repository isn't available. Does it makes sense to keep this PR?

**Comment by tdonohue** [comment]
Closing as this PR seems outdated & unmaintained and it seems to be replaced by #2987

**Pull request closed without merging**